scripts/prepare_docker_buildinfo.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+IMAGENAME=$1
+DOCKERFILE=$2
+ARCH=$3
+DOCKERFILE_TARGE=$4
+DISTRO=$5
+
+[ -z "$BUILD_SLAVE" ] && BUILD_SLAVE=n
+[ -z "$DOCKERFILE_TARGE" ] && DOCKERFILE_TARGE=$DOCKERFILE
+DOCKERFILE_PATH=$(dirname "$DOCKERFILE_TARGE")
+BUILDINFO_PATH="${DOCKERFILE_PATH}/buildinfo"
+BUILDINFO_VERSION_PATH="${BUILDINFO_PATH}/versions"
+
+[ -d $BUILDINFO_PATH ] && rm -rf $BUILDINFO_PATH
+mkdir -p $BUILDINFO_VERSION_PATH
+
+# Get the debian distribution from the docker base image
+if [ -z "$DISTRO" ]; then
+    DOCKER_BASE_IMAGE=$(grep "^FROM" $DOCKERFILE | head -n 1 | awk '{print $2}')
+    DISTRO=$(docker run --rm --entrypoint "" $DOCKER_BASE_IMAGE cat /etc/os-release | grep VERSION_CODENAME | cut -d= -f2)
+    [ -z "$DISTRO" ] && DISTRO=jessie
+fi
+
+DOCKERFILE_PRE_SCRIPT='# Auto-Generated for buildinfo 
```

Comment:
>DOCKERFILE_PRE_SCRIPT [](start = 0, length = 21)

If `gpg` comment is right, then `DOCKERFILE_PRE_SCRIPT` has only 2 command lines. Suggest move them into docker-base, docker-slave's Dockerfile as plaintext. No need to manipulate Dockerfile here.

Reply:
See the comments below in pending.

---
In reply to: [539816717](https://github.com/Azure/sonic-buildimage/pull/5718#discussion_r539816717) [](ancestors = 539816717)